src/Components/UI/Apexcharts/index.js

```diff
@@ -0,0 +1,283 @@
+      text: "Financing Activities"
+    },
+    {
+      id: 3,
+      name: "ncfia",
+      text: "Investing Activities"
+    },
+    {
+      id: 4,
+      name: "fcf",
+      text: "Free cash flow"
+    }
+  ];
+  const fetchIncomeData = async () => {
+    let tag;
+    if (index === 1) {
+      tag = "netcashfromoperatingactivities";
+    }
+    if (index === 2) {
+      tag = "netcashfromfinancingactivities";
+    }
+    if (index === 3) {
+      tag = "netcashfrominvestingactivities";
+    }
+    if (index === 4) {
+      tag = "freecashflow";
+    }
+    const responseCashChart = await axios.get(
+      `http://ec2-3-84-165-185.compute-1.amazonaws.com:8080/SpringJDBCApp-0.0.1-SNAPSHOT/stocks/cashflowgraph?ticker=${tickerParam}`
+    );
+    const cashChartResponse = responseCashChart.data.reverse();
+    const uiCashChartData = cashChartResponse.map(item => {
+      return {
+        x: new Date(item.fiscal_year.toString()),
+        y: item[`${tag}`]
+      };
+    });
+    if (index === 4) {
+      try {
+        const responseForecast = await axios.get(
+          `http://ec2-35-175-187-45.compute-1.amazonaws.com/revenue/forecast/ticker=${tickerParam}`
+        );
+        let cashChartInfo= typeof(responseForecast.data) === "string" ? JSON.parse(responseForecast.data.replace(/NaN/g, 0)) : responseForecast.data
+        const uiFutureData = [
+          {
+            x: new Date(
+              (
+                cashChartResponse[cashChartResponse.length - 1]
+                  .fiscal_year + 1
+              ).toString()
+            ),
+            y: cashChartInfo[`free_cash_flow_0_yrs`]
+          },
+          {
+            x: new Date(
+              (
+                cashChartResponse[cashChartResponse.length - 1]
+                  .fiscal_year + 2
+              ).toString()
+            ),
+            y: cashChartInfo[`free_cash_flow_1_yrs`]
+          },
+          {
+            x: new Date(
+              (
+                cashChartResponse[cashChartResponse.length - 1]
+                  .fiscal_year + 3
+              ).toString()
+            ),
+            y: cashChartInfo[`free_cash_flow_2_yrs`]
+          }
+        ];
+        let newData = uiCashChartData.concat(uiFutureData);
+        setCashFlowData(newData);
+      } catch (err) {
+        setCashFlowData(uiCashChartData);
+      }
+    } else {
+      setCashFlowData(uiCashChartData);
+    }
+  };
+  useEffect(() => {
+    fetchIncomeData();
+  }, [index]);
+  const handleIncomeChart = async event => {
+    const { name } = event.target;
+    switch (name) {
+      case "ncffa":
+        setIndex(2);
+        setToolData("Financing Activities");
+        break;
+      case "ncfia":
+        setIndex(3);
+        setToolData("Investing Activities");
+        break;
+      case "fcf":
+        setIndex(4);
+        setToolData("Free cash flow");
+        break;
+      default:
+        setIndex(1);
+        setToolData("Operating Activities");
+    }
```

Comment:
We can handle this using one object . create a object with key Name , index and value string